### PR TITLE
Rewrite chrome URLs to compare them.

### LINKED ISSUE #1776: GatherRunner:error timeout reached for chrome://settings
When testing Chrome tip of tree with --chrome-flags="--disable-features=MaterialDesignSettings", or testing Stable without the above flag (it's disabled by default on Stable), chrome://settings fails to run.

This is with lighthouse also on master/1.5.1.

Full command:

```sh
lighthouse --chrome-flags="--disable-features=MaterialDesignSettings" chrome://settings
```

### PR DESCRIPTION
**Pull request by paulirish**
fixes #1776

### CODE REVIEW

**Comment by paulirish** [comment]
@tommycli can you review?

**Comment by tommycli** [comment]
Btw, I don't think I'm a valid reviewer.

File: lighthouse-core/lib/url-shim.js [hunk 1]
```diff
@@ -94,13 +94,24 @@ URL.getDisplayName = function getDisplayName(url) {
   return name;
 };
 
+// There is fancy URL rewriting logic for the chrome://settings page that we need to work around.
+// Why? It was to be able to swap from chrome://settings to chrome://extensions with a JavaScript
+// pushState call without forcing a page reload. :)
+// As a result, the final document URL doesn't match the URL that network layer surfaces.
+function rewriteChromeInternalUrl(url) {
+  if (!url.startsWith('chrome://')) return url;
+  return url.replace(/^chrome:\/\/chrome\//, 'chrome://');
```
Comment (ebidel) [inline lighthouse-core/lib/url-shim.js]: it't not clear from your description that it uses `chrome:chrome` urls. Might want to update the summary.

Comment (paulirish) [inline lighthouse-core/lib/url-shim.js]: good call. you got it.

**Pull request merged**